plugins/modules/vmware_guest.py

```diff
@@ -1757,14 +1756,9 @@ def configure_hardware_params(self, vm_obj):
                 try:
                     temp_version = int(temp_version)
                 except ValueError:
-                    hw_version_check_failed = True
-
-                if temp_version not in range(3, 18):
-                    hw_version_check_failed = True
-
-                if hw_version_check_failed:
                     self.module.fail_json(msg="Failed to set hardware.version '%s' value as valid"
-                                          " values range from 3 (ESX 2.x) to 17 (ESXi 7.0)." % temp_version)
+                                          " values are either 'latest' or a number." % temp_version)
```

Comment:
@mariolenz Thanks for PR. Could you please add a note `version` documentation about range of valid number?

Reply:
No, because documenting the range of valid numbers would mean we have to change this every time VMware invents a new version number. But the documentation check the VMware documentation for valid version numbers and I've included a similar statement in this error message.